Encoding issues after using Powershell 7.4.0

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

When using Oh My Posh, which was successfully working just fine with Powershell 7.3.10, after upgrading to 7.4.0 all text becomes unreadable. See here for example:

![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/6610039/9cd35ee5-cf8e-4a0a-aab8-f487e8c6a8df)

The fix it turns out was here: https://github.com/PowerShell/PowerShell/issues/19941

```ps
[Console]::OutputEncoding = [Text.Encoding]::UTF8

try {
    oh-my-posh init pwsh --config ~/oscar.omp.json | Invoke-Expression
} finally {
}
```

The key part being [Console]::OutputEncoding = [Text.Encoding]::UTF8

After implementing that fix, it works correctly:

![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/6610039/12d0a804-92d2-45e6-b9dc-2caff154e806)

I am not sure if this is something that Oh My Posh can accomidate without doing the above "work around" or if the documentation now needs to be updated to indicate this line is now required with Powershell 7.4.0 and beyond: [Console]::OutputEncoding = [Text.Encoding]::UTF8

### Theme

Atomic

### What OS are you seeing the problem on?

Windows

### Which shell are you using?

powershell

### Log output

```Shell
Version: 18.26.1

Shell: pwsh (7.4.0)

Prompt:

╭─ pwsh  C:\Temp  0ms⠀                                                                  99   28,15:38 
╰─

Segments:

ConsoleTitle(false)                        -   0 ms
Shell(true)                                -   1 ms
Root(false)                                -   1 ms
Path(true)                                 -   2 ms
Git(false)                                 -  21 ms
Executiontime(true)                        -   1 ms
Node(false)                                -   6 ms
Python(false)                              -   5 ms
Java(false)                                -   6 ms
Dotnet(false)                              -  10 ms
Go(false)                                  -   4 ms
Rust(false)                                -   5 ms
Dart(false)                                -   6 ms
Angular(false)                             -   2 ms
Nx(false)                                  -   4 ms
Julia(false)                               -   4 ms
Ruby(false)                                -   4 ms
Azfunc(false)                              -   3 ms
Aws(false)                                 -   0 ms
Kubectl(false)                             -  88 ms
Os(true)                                   -   0 ms
Battery(true)                              -   7 ms
Time(true)                                 -   0 ms
Text(true)                                 -   0 ms
Status(true)                               -   0 ms

Run duration: 124.6436ms

Cache path: C:\Users\matto\AppData\Local\oh-my-posh

Config path: C:\TFS\TerminalConfig\atomic.omp.json

Logs:

[DEBUG] 15:38:05.136 shell.go:Getenv:398 → NO DATA
[TRACE] 15:38:05.136 shell.go:Getenv(TERM_PROGRAM) - 502.5µs
[DEBUG] 15:38:05.136 shell.go:Getenv:398 → C:\Users\matto\AppData\Local
[TRACE] 15:38:05.136 shell.go:Getenv(LOCALAPPDATA) - 45.2µs
[TRACE] 15:38:05.136 shell_windows.go:CachePath() - 45.2µs
[DEBUG] 15:38:05.136 shell.go:Getenv:398 → C:\TFS\TerminalConfig\atomic.omp.json
[TRACE] 15:38:05.136 shell.go:Getenv(POSH_THEME) - 0s
[DEBUG] 15:38:05.136 shell.go:Shell:652 → no shell name provided in flags, trying to detect it
[DEBUG] 15:38:05.142 shell.go:Shell:660 → process name: pwsh.exe
[TRACE] 15:38:05.142 shell.go:Shell() - 6.2565ms
[TRACE] 15:38:05.142 shell.go:resolveConfigPath() - 6.2565ms
[TRACE] 15:38:05.143 shell.go:Init() - 6.8042ms
[TRACE] 15:38:05.143 shell.go:Flags() - 0s
[TRACE] 15:38:05.144 config.go:loadConfig() - 1.5856ms
[TRACE] 15:38:05.144 shell.go:Flags() - 0s
[DEBUG] 15:38:05.144 shell.go:Getenv:398 → NO DATA
[TRACE] 15:38:05.144 shell.go:Getenv(OMP_CACHE_DISABLED) - 0s
[TRACE] 15:38:05.144 shell_windows.go:WindowsRegistryKeyValue(HKEY_CURRENT_USER\Software\Microsoft\Windows\DWM\ColorizationColor) - 0s
[TRACE] 15:38:05.144 shell.go:GOOS() - 0s
[TRACE] 15:38:05.144 shell.go:GOOS() - 0s
[TRACE] 15:38:05.144 shell.go:GOOS() - 0s
[TRACE] 15:38:05.144 shell.go:GOOS() - 0s
[TRACE] 15:38:05.144 shell.go:GOOS() - 0s
[TRACE] 15:38:05.144 shell.go:GOOS() - 0s
[TRACE] 15:38:05.144 shell.go:GOOS() - 0s
[TRACE] 15:38:05.144 shell.go:GOOS() - 0s
[TRACE] 15:38:05.144 shell.go:GOOS() - 0s
[TRACE] 15:38:05.144 shell.go:GOOS() - 0s
[TRACE] 15:38:05.144 shell.go:GOOS() - 0s
[TRACE] 15:38:05.144 shell.go:GOOS() - 0s
[TRACE] 15:38:05.144 shell.go:GOOS() - 0s
[TRACE] 15:38:05.144 shell.go:GOOS() - 0s
[TRACE] 15:38:05.144 shell.go:GOOS() - 0s
[TRACE] 15:38:05.144 shell.go:GOOS() - 0s
[TRACE] 15:38:05.144 shell.go:GOOS() - 0s
[TRACE] 15:38:05.144 shell.go:GOOS() - 0s
[TRACE] 15:38:05.144 shell.go:GOOS() - 0s
[DEBUG] 15:38:05.144 shell_windows.go:WindowsRegistryKeyValue:215 → ColorizationColor(DWORD): 0xC40078D7
[TRACE] 15:38:05.144 shell.go:Shell() - 0s
[DEBUG] 15:38:05.144 shell.go:Getenv:398 → 7.4.0
[TRACE] 15:38:05.144 shell.go:Getenv(POSH_SHELL_VERSION) - 0s
[DEBUG] 15:38:05.144 debug.go:PrintDebug:22 → Segment: Title
[DEBUG] 15:38:05.144 text.go:Render:70 → Rendering template:
[DEBUG] 15:38:05.144 shell.go:Getenv:398 → 6
[TRACE] 15:38:05.144 shell.go:Getenv(POSH_CURSOR_LINE) - 0s
[DEBUG] 15:38:05.144 shell.go:Getenv:398 → 1
[TRACE] 15:38:05.144 shell.go:Getenv(POSH_CURSOR_COLUMN) - 0s
[TRACE] 15:38:05.144 shell.go:Flags() - 0s
[TRACE] 15:38:05.144 shell.go:Flags() - 0s
[TRACE] 15:38:05.145 shell.go:Shell() - 0s
[TRACE] 15:38:05.145 shell.go:Shell() - 0s
[TRACE] 15:38:05.145 shell.go:Flags() - 0s
[TRACE] 15:38:05.145 shell.go:Flags() - 0s
[TRACE] 15:38:05.145 shell.go:GOOS() - 0s
[TRACE] 15:38:05.145 shell.go:Flags() - 0s
[TRACE] 15:38:05.145 shell.go:Flags() - 0s
[DEBUG] 15:38:05.145 shell.go:Pwd:428 → C:\Temp
[TRACE] 15:38:05.145 shell.go:Pwd() - 0s
[TRACE] 15:38:05.145 shell.go:Flags() - 0s
[DEBUG] 15:38:05.145 segment.go:SetEnabled:512 → Segment: Executiontime
[DEBUG] 15:38:05.145 properties.go:GetBool:22 → always_enabled: false
[TRACE] 15:38:05.145 shell.go:ExecutionTime() - 0s
[DEBUG] 15:38:05.145 properties.go:GetFloat64:34 → threshold: 0.000000
[DEBUG] 15:38:05.145 properties.go:GetString:28 → roundrock
[TRACE] 15:38:05.145 shell_windows.go:Root() - 0s
[TRACE] 15:38:05.145 shell.go:Shell() - 0s
[TRACE] 15:38:05.145 shell.go:StatusCodes() - 0s
[TRACE] 15:38:05.145 shell_windows.go:IsWsl() - 0s
[TRACE] 15:38:05.145 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.145 shell_windows.go:56 → C:\Users\matto
[TRACE] 15:38:05.145 shell.go:GOOS() - 0s
[DEBUG] 15:38:05.145 segment.go:SetEnabled:512 → Segment: Git
[TRACE] 15:38:05.145 shell.go:GOOS() - 0s
[TRACE] 15:38:05.146 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.146 shell_windows.go:56 → C:\Users\matto
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[DEBUG] 15:38:05.146 segment.go:SetEnabled:512 → Segment: Shell
[TRACE] 15:38:05.146 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.146 properties.go:GetKeyValueMap:46 → mapped_shell_names: map[]
[DEBUG] 15:38:05.146 shell_windows.go:56 → C:\Users\matto
[TRACE] 15:38:05.146 shell.go:Shell() - 0s
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[TRACE] 15:38:05.146 shell.go:Flags() - 0s
[TRACE] 15:38:05.146 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.146 segment.go:SetEnabled:512 → Segment: Root
[DEBUG] 15:38:05.146 shell_windows.go:56 → C:\Users\matto
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[DEBUG] 15:38:05.146 segment.go:SetEnabled:512 → Segment: Path
[TRACE] 15:38:05.146 shell.go:Pwd() - 0s
[TRACE] 15:38:05.146 shell_windows.go:Root() - 0s
[TRACE] 15:38:05.146 shell.go:Shell() - 0s
[TRACE] 15:38:05.146 shell.go:Flags() - 0s
[DEBUG] 15:38:05.146 shell.go:TemplateCache:833 → environment: [ALLUSERSPROFILE=C:\ProgramData APPDATA=C:\Users\matto\AppData\Roaming asl.log=Destination=file ASPNETCORE_ENVIRONMENT=Development AZURE_CLIENT_ID=d0d4d4d3-d817-479b-910b-1254482326ad AZURE_CLIENT_SECRET=4CZz2dLdrXy3tR3x5-Cx4PN8SC.1y_gS4W AZURE_TENANT_ID=51c99a9c-bb9a-4dc7-ac91-464d50cc8d1f ChocolateyInstall=C:\ProgramData\chocolatey ChocolateyLastPathUpdate=132730111847777415 CLASSPATH=C:\PROGRA~1\IBM\SQLLIB\bin CLIENTNAME=MATT-PC CommonProgramFiles=C:\Program Files\Common Files CommonProgramFiles(x86)=C:\Program Files (x86)\Common Files CommonProgramW6432=C:\Program Files\Common Files COMPUTERNAME=MATTOLSON-LAP ComSpec=C:\Windows\system32\cmd.exe CONDA_PROMPT_MODIFIER=False CPCV2.22.7.68_Home=C:\Program Files (x86)\Pearson VUE\Compiler\ DB2INSTANCE=DB2 DriverData=C:\Windows\System32\Drivers\DriverData envContainerTelemetryApiCmdLine=-st "C:\Program Files\NVIDIA Corporation\NvContainer\NvContainerTelemetryApi.dll" envContainerTelemetryApiCmdLineX86=-st "C:\Program Files (x86)\NVIDIA Corporation\NvContainer\NvContainerTelemetryApi.dll" HOMEDRIVE=C: HOMEPATH=\Users\matto INCLUDE=C:\PROGRA~1\IBM\SQLLIB\LIB LIB=;C:\PROGRA~1\IBM\SQLLIB\LIB LOCALAPPDATA=C:\Users\matto\AppData\Local LOGONSERVER=\\JOSEPHINE NUMBER_OF_PROCESSORS=16 OneDrive=C:\Users\matto\OneDrive -  OneDriveCommercial=C:\Users\matto\OneDrive -  OS=Windows_NT Path=C:\Program Files\PowerShell\7;C:\Program Files\Python39\Scripts\;C:\Program Files\Python39\;C:\Program Files\Python310\Scripts\;C:\Program Files\Python310\;C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files\Microsoft VS Code\bin;C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\dotnet\;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn\;C:\Program Files\Microsoft\Web Platform Installer\;C:\Program Files (x86)\Microsoft SQL Server\150\DTS\Binn\;C:\Program Files\Azure Data Studio\bin;C:\Program Files (x86)\Microsoft SQL Server\110\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\120\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\130\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\140\DTS\Binn\;C:\Program Files (x86)\IBM\Client Access\Shared;C:\Program Files (x86)\IBM\Client Access\;C:\Program Files\Seq\;C:\Program Files\Seq\Client\;C:\Program Files\ibm\gsk8\lib64;C:\Program Files (x86)\ibm\gsk8\lib;C:\PROGRA~1\IBM\SQLLIB\BIN;C:\PROGRA~1\IBM\SQLLIB\FUNCTION;C:\ProgramData\chocolatey\bin;C:\Program Files\Microsoft SQL Server\150\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\160\DTS\Binn\;C:\Program Files\nodejs\;C:\Program Files\Git\cmd;C:\Program Files (x86)\GitExtensions\;C:\Program Files\Microsoft SQL Server\140\DTS\Binn\;C:\Program Files\Microsoft SQL Server\150\DTS\Binn\;C:\Program Files\Microsoft SQL Server\160\DTS\Binn\;C:\Program Files\Docker\Docker\resources\bin;C:\Program Files\Microsoft SQL Server\130\DTS\Binn\;C:\Program Files\PowerShell\7\;C:\Users\matto\AppData\Local\Microsoft\WindowsApps;C:\Users\matto\.dotnet\tools;C:\Users\matto\.dotnet\tools;C:\Users\matto\AppData\Roaming\npm;;C:\Users\matto\AppData\Local\Programs\oh-my-posh\bin PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.PY;.PYW;.CPL POSH_AZURE_ENABLED=False POSH_CURSOR_COLUMN=1 POSH_CURSOR_LINE=6 POSH_GIT_ENABLED=False POSH_INSTALLER=winget POSH_PID=4764 POSH_SHELL_VERSION=7.4.0 POSH_THEME=C:\TFS\TerminalConfig\atomic.omp.json POSH_THEMES_PATH=C:\Users\matto\AppData\Local\Programs\oh-my-posh\themes POWERLINE_COMMAND=oh-my-posh POWERSHELL_DISTRIBUTION_CHANNEL=MSI:Windows 10 Enterprise POWERSHELL_TELEMETRY_OPTOUT=1 PROCESSOR_ARCHITECTURE=AMD64 PROCESSOR_IDENTIFIER=Intel64 Family 6 Model 165 Stepping 2, GenuineIntel PROCESSOR_LEVEL=6 PROCESSOR_REVISION=a502 ProgramData=C:\ProgramData ProgramFiles=C:\Program Files ProgramFiles(x86)=C:\Program Files (x86) ProgramW6432=C:\Program Files PSModulePath=C:\Users\matto\OneDrive - \Documents\PowerShell\Modules;C:\Program Files\PowerShell\Modules;c:\program files\powershell\7\Modules;C:\Program Files\WindowsPowerShell\Modules;C:\Windows\system32\WindowsPowerShell\v1.0\Modules PUBLIC=C:\Users\Public SESSIONNAME=RDP-Tcp#2 SystemDrive=C: SystemRoot=C:\Windows TEMP=C:\Users\matto\AppData\Local\Temp TMP=C:\Users\matto\AppData\Local\Temp USERDNSDOMAIN=NT.ORG USERDOMAIN=NT USERDOMAIN_ROAMINGPROFILE=NT USERNAME=matto USERPROFILE=C:\Users\matto windir=C:\Windows WSLENV=WT_SESSION:WT_PROFILE_ID: WT_PROFILE_ID={574e775e-4f2a-5b96-ac1e-a2962a402336} WT_SESSION=adaa30cb-80fc-443c-8680-b5bc98306362 ZES_ENABLE_SYSMAN=1]
[DEBUG] 15:38:05.146 properties.go:GetBool:22 → mapped_locations_enabled: true
[DEBUG] 15:38:05.146 properties.go:GetString:28 → 
[DEBUG] 15:38:05.146 properties.go:GetString:28 → 
[DEBUG] 15:38:05.146 shell_windows.go:56 → C:\Users\matto
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[TRACE] 15:38:05.146 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.146 properties.go:GetString:28 → 
[DEBUG] 15:38:05.146 properties.go:GetKeyValueMap:46 → mapped_locations: map[]
[DEBUG] 15:38:05.146 shell_windows.go:56 → C:\Users\matto
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[TRACE] 15:38:05.146 shell.go:PathSeparator() - 0s
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[DEBUG] 15:38:05.146 shell.go:User:563 → matto
[TRACE] 15:38:05.146 shell.go:User() - 0s
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[DEBUG] 15:38:05.146 shell.go:Host:575 → MattOlson-Lap
[TRACE] 15:38:05.146 shell.go:Host() - 0s
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[DEBUG] 15:38:05.146 shell.go:Getenv:398 → NO DATA
[TRACE] 15:38:05.146 shell.go:Getenv(SHLVL) - 0s
[TRACE] 15:38:05.146 shell.go:TemplateCache() - 1.0727ms
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[TRACE] 15:38:05.146 shell.go:PathSeparator() - 0s
[TRACE] 15:38:05.146 shell.go:TemplateCache() - 0s
[TRACE] 15:38:05.146 shell.go:PathSeparator() - 0s
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[TRACE] 15:38:05.146 shell.go:PathSeparator() - 0s
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[TRACE] 15:38:05.146 shell.go:PathSeparator() - 0s
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[TRACE] 15:38:05.146 shell.go:GOOS() - 0s
[TRACE] 15:38:05.146 shell.go:PathSeparator() - 0s
[DEBUG] 15:38:05.146 properties.go:GetString:28 → full
[TRACE] 15:38:05.146 shell.go:PathSeparator() - 0s
[DEBUG] 15:38:05.146 properties.go:GetStringArray:52 → cycle: []
[DEBUG] 15:38:05.146 properties.go:GetString:28 → NO DATA
[TRACE] 15:38:05.146 shell.go:PathSeparator() - 0s
[DEBUG] 15:38:05.146 properties.go:GetString:28 → \
[DEBUG] 15:38:05.146 properties.go:GetBool:22 → cycle_folder_separator: false
[DEBUG] 15:38:05.146 properties.go:GetString:28 → %s
[DEBUG] 15:38:05.146 properties.go:GetString:28 → %s
[TRACE] 15:38:05.146 shell.go:PathSeparator() - 0s
[TRACE] 15:38:05.146 shell.go:Pwd() - 0s
[TRACE] 15:38:05.146 shell_windows.go:IsWsl() - 0s
[TRACE] 15:38:05.147 shell.go:StackCount() - 0s
[DEBUG] 15:38:05.147 win32_windows.go:isWriteable:303 → current user is member of S-1-5-32-544
[DEBUG] 15:38:05.147 win32_windows.go:isWriteable:311 ↓
    WRITE_DAC
    WRITE_OWNER
    SYNCHRONIZE
    DELETE
    READ_CONTROL
[DEBUG] 15:38:05.147 win32_windows.go:isWriteable:313 → user has write access
[TRACE] 15:38:05.147 shell_windows.go:DirIsWritable() - 518.2µs
[TRACE] 15:38:05.147 shell.go:TemplateCache() - 0s
[DEBUG] 15:38:05.166 shell.go:CommandPath:614 → C:\Program Files\Git\cmd\git.exe
[TRACE] 15:38:05.166 shell.go:CommandPath(git.exe) - 20.834ms
[TRACE] 15:38:05.166 shell.go:HasCommand(git.exe) - 20.834ms
[TRACE] 15:38:05.166 shell.go:Pwd() - 0s
[ERROR] 15:38:05.167 shell.go:HasParentFilePath:748 → CreateFile .git: The system cannot find the file specified.
[TRACE] 15:38:05.167 shell.go:HasParentFilePath(.git) - 518.6µs
[DEBUG] 15:38:05.167 properties.go:GetBool:22 → fetch_bare_info: false
[DEBUG] 15:38:05.167 text.go:Render:70 → Rendering template:  {{ .Name }}
[DEBUG] 15:38:05.167 text.go:Render:70 → Rendering template:  {{ .FormattedMs }}⠀
[DEBUG] 15:38:05.167 text.go:Render:70 → Rendering template:  {{ .Path }}
[TRACE] 15:38:05.167 shell.go:TemplateCache() - 0s
[TRACE] 15:38:05.167 shell.go:TemplateCache() - 0s
[TRACE] 15:38:05.167 shell.go:Shell() - 0s
[TRACE] 15:38:05.167 shell.go:Shell() - 0s
[TRACE] 15:38:05.167 shell.go:TemplateCache() - 0s
[TRACE] 15:38:05.167 shell.go:Shell() - 0s
[DEBUG] 15:38:05.167 text.go:Render:70 → Rendering template: diamond
[DEBUG] 15:38:05.167 text.go:Render:70 → Rendering template: diamond
[DEBUG] 15:38:05.167 text.go:Render:70 → Rendering template: powerline
[DEBUG] 15:38:05.167 text.go:Render:70 → Rendering template: powerline
[DEBUG] 15:38:05.167 text.go:Render:70 → Rendering template: diamond
[TRACE] 15:38:05.167 shell.go:Shell() - 0s
[TRACE] 15:38:05.167 shell.go:Shell() - 0s
[TRACE] 15:38:05.167 shell.go:Shell() - 0s
[TRACE] 15:38:05.167 shell.go:Flags() - 0s
[TRACE] 15:38:05.167 shell.go:Flags() - 0s
[TRACE] 15:38:05.167 shell.go:Pwd() - 0s
[TRACE] 15:38:05.167 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.167 shell_windows.go:56 → C:\Users\matto
[TRACE] 15:38:05.167 shell.go:GOOS() - 0s
[DEBUG] 15:38:05.167 segment.go:SetEnabled:512 → Segment: Node
[DEBUG] 15:38:05.167 shell_windows.go:56 → C:\Users\matto
[TRACE] 15:38:05.167 shell.go:GOOS() - 0s
[DEBUG] 15:38:05.167 segment.go:SetEnabled:512 → Segment: Time
[DEBUG] 15:38:05.167 properties.go:GetStringArray:52 → extensions: [*.js *.ts package.json .nvmrc pnpm-workspace.yaml .pnpmfile.cjs .npmrc .vue]
[TRACE] 15:38:05.167 shell.go:Flags() - 0s
[DEBUG] 15:38:05.167 properties.go:GetStringArray:52 → folders: []
[DEBUG] 15:38:05.167 properties.go:GetBool:22 → home_enabled: false
[TRACE] 15:38:05.167 shell.go:Pwd() - 0s
[TRACE] 15:38:05.167 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.167 properties.go:GetString:28 → _2,15:04
[DEBUG] 15:38:05.167 shell_windows.go:56 → C:\Users\matto
[TRACE] 15:38:05.167 shell.go:TemplateCache() - 0s
[DEBUG] 15:38:05.167 properties.go:GetString:28 → files
[TRACE] 15:38:05.167 shell.go:Flags() - 0s
[DEBUG] 15:38:05.167 shell_windows.go:56 → C:\Users\matto
[TRACE] 15:38:05.167 shell.go:GOOS() - 0s
[DEBUG] 15:38:05.167 properties.go:GetBool:22 → fetch_package_manager: true
[TRACE] 15:38:05.167 shell.go:Flags() - 0s
[DEBUG] 15:38:05.167 segment.go:SetEnabled:512 → Segment: Nx
[TRACE] 15:38:05.167 shell.go:Pwd() - 0s
[TRACE] 15:38:05.167 shell.go:Pwd() - 0s
[TRACE] 15:38:05.167 shell.go:Flags() - 0s
[DEBUG] 15:38:05.167 shell_windows.go:56 → C:\Users\matto
[DEBUG] 15:38:05.167 properties.go:GetStringArray:52 → extensions: [workspace.json nx.json]
[TRACE] 15:38:05.167 shell.go:GOOS() - 0s
[TRACE] 15:38:05.167 shell.go:Pwd() - 0s
[TRACE] 15:38:05.168 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.168 shell_windows.go:56 → C:\Users\matto
[TRACE] 15:38:05.168 shell.go:GOOS() - 0s
[DEBUG] 15:38:05.168 segment.go:SetEnabled:512 → Segment: Dart
[DEBUG] 15:38:05.168 shell_windows.go:56 → C:\Users\matto
[DEBUG] 15:38:05.168 properties.go:GetStringArray:52 → extensions: [*.dart pubspec.yaml pubspec.yml pubspec.lock]
[DEBUG] 15:38:05.168 properties.go:GetString:28 → files
[DEBUG] 15:38:05.168 properties.go:GetStringArray:52 → folders: [.dart_tool]
[TRACE] 15:38:05.168 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.168 properties.go:GetBool:22 → home_enabled: false
[TRACE] 15:38:05.168 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.168 shell_windows.go:56 → C:\Users\matto
[DEBUG] 15:38:05.168 properties.go:GetString:28 → files
[TRACE] 15:38:05.168 shell.go:Pwd() - 0s
[TRACE] 15:38:05.168 shell.go:Flags() - 0s
[TRACE] 15:38:05.168 shell.go:Flags() - 0s
[DEBUG] 15:38:05.168 properties.go:GetString:28 → environment
[TRACE] 15:38:05.168 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.168 shell_windows.go:56 → C:\Users\matto
[TRACE] 15:38:05.168 shell.go:GOOS() - 0s
[DEBUG] 15:38:05.168 segment.go:SetEnabled:512 → Segment: Python
[DEBUG] 15:38:05.168 shell.go:Getenv:398 → NO DATA
[DEBUG] 15:38:05.168 properties.go:GetStringArray:52 → extensions: [*.py *.ipynb pyproject.toml venv.bak]
[TRACE] 15:38:05.168 shell.go:Getenv(JAVA_HOME) - 0s
[DEBUG] 15:38:05.168 properties.go:GetStringArray:52 → folders: [.venv venv virtualenv venv-win pyenv-win]
[TRACE] 15:38:05.168 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.168 properties.go:GetBool:22 → home_enabled: false
[DEBUG] 15:38:05.168 shell_windows.go:56 → C:\Users\matto
[TRACE] 15:38:05.168 shell.go:Flags() - 0s
[TRACE] 15:38:05.168 shell.go:GOOS() - 0s
[TRACE] 15:38:05.168 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.168 segment.go:SetEnabled:512 → Segment: Java
[DEBUG] 15:38:05.168 shell_windows.go:56 → C:\Users\matto
[TRACE] 15:38:05.168 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.168 properties.go:GetBool:22 → fetch_virtual_env: true
[DEBUG] 15:38:05.168 shell_windows.go:56 → C:\Users\matto
[DEBUG] 15:38:05.168 properties.go:GetStringArray:52 → extensions: [pom.xml build.gradle.kts build.sbt .java-version .deps.edn project.clj build.boot *.java *.class *.gradle *.jar *.clj *.cljc]
[TRACE] 15:38:05.168 shell.go:GOOS() - 0s
[DEBUG] 15:38:05.168 properties.go:GetStringArray:52 → folders: []
[DEBUG] 15:38:05.168 properties.go:GetBool:22 → home_enabled: false
[DEBUG] 15:38:05.168 segment.go:SetEnabled:512 → Segment: Ruby
[TRACE] 15:38:05.168 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.168 shell_windows.go:56 → C:\Users\matto
[DEBUG] 15:38:05.168 properties.go:GetString:28 → files
[DEBUG] 15:38:05.168 properties.go:GetStringArray:52 → extensions: [*.rb Rakefile Gemfile]
[TRACE] 15:38:05.168 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.168 properties.go:GetStringArray:52 → folders: []
[DEBUG] 15:38:05.168 properties.go:GetBool:22 → home_enabled: false
[TRACE] 15:38:05.168 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.168 shell_windows.go:56 → C:\Users\matto
[DEBUG] 15:38:05.168 properties.go:GetString:28 → files
[TRACE] 15:38:05.168 shell.go:Pwd() - 0s
[TRACE] 15:38:05.168 shell.go:Flags() - 0s
[TRACE] 15:38:05.168 shell.go:Flags() - 0s
[TRACE] 15:38:05.168 shell.go:Pwd() - 0s
[TRACE] 15:38:05.168 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.168 shell_windows.go:56 → C:\Users\matto
[DEBUG] 15:38:05.168 shell_windows.go:56 → C:\Users\matto
[TRACE] 15:38:05.168 shell.go:GOOS() - 0s
[DEBUG] 15:38:05.168 segment.go:SetEnabled:512 → Segment: Angular
[DEBUG] 15:38:05.168 properties.go:GetStringArray:52 → extensions: [angular.json]
[DEBUG] 15:38:05.168 properties.go:GetStringArray:52 → folders: []
[DEBUG] 15:38:05.168 properties.go:GetBool:22 → home_enabled: false
[TRACE] 15:38:05.168 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.168 shell_windows.go:56 → C:\Users\matto
[DEBUG] 15:38:05.168 properties.go:GetString:28 → files
[TRACE] 15:38:05.168 shell.go:Pwd() - 0s
[TRACE] 15:38:05.168 shell.go:GOOS() - 0s
[DEBUG] 15:38:05.168 shell_windows.go:56 → C:\Users\matto
[TRACE] 15:38:05.168 shell.go:GOOS() - 0s
[DEBUG] 15:38:05.168 segment.go:SetEnabled:512 → Segment: Go
[TRACE] 15:38:05.168 shell.go:Flags() - 0s
[TRACE] 15:38:05.168 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.168 properties.go:GetStringArray:52 → extensions: [*.go go.mod]
[DEBUG] 15:38:05.168 properties.go:GetStringArray:52 → folders: []
[DEBUG] 15:38:05.168 properties.go:GetBool:22 → home_enabled: false
[TRACE] 15:38:05.168 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.168 shell_windows.go:56 → C:\Users\matto
[DEBUG] 15:38:05.168 properties.go:GetString:28 → files
[TRACE] 15:38:05.168 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.168 segment.go:SetEnabled:512 → Segment: Dotnet
[DEBUG] 15:38:05.168 shell_windows.go:56 → C:\Users\matto
[TRACE] 15:38:05.168 shell.go:GOOS() - 0s
[DEBUG] 15:38:05.168 segment.go:SetEnabled:512 → Segment: Rust
[TRACE] 15:38:05.168 shell.go:Flags() - 0s
[DEBUG] 15:38:05.168 properties.go:GetStringArray:52 → extensions: [*.cs *.csx *.vb *.sln *.slnf *.csproj *.vbproj *.fs *.fsx *.fsproj global.json]
[DEBUG] 15:38:05.168 properties.go:GetStringArray:52 → folders: []
[TRACE] 15:38:05.168 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.168 properties.go:GetBool:22 → home_enabled: false
[TRACE] 15:38:05.168 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.168 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.168 shell.go:HasFiles(yarn.lock) - 1.2309ms
[TRACE] 15:38:05.168 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.169 properties.go:GetStringArray:52 → extensions: [*.rs Cargo.toml Cargo.lock]
[DEBUG] 15:38:05.169 properties.go:GetStringArray:52 → folders: []
[DEBUG] 15:38:05.169 properties.go:GetBool:22 → home_enabled: false
[TRACE] 15:38:05.169 shell.go:Pwd() - 516.8µs
[DEBUG] 15:38:05.169 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.169 shell.go:HasFiles(*.rb) - 1.6902ms
[DEBUG] 15:38:05.169 shell_windows.go:56 → C:\Users\matto
[TRACE] 15:38:05.169 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.169 shell_windows.go:56 → C:\Users\matto
[TRACE] 15:38:05.169 shell.go:GOOS() - 0s
[DEBUG] 15:38:05.169 segment.go:SetEnabled:512 → Segment: Kubectl
[DEBUG] 15:38:05.169 properties.go:GetBool:22 → parse_kubeconfig: false
[DEBUG] 15:38:05.170 properties.go:GetString:28 → files
[TRACE] 15:38:05.170 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.170 shell_windows.go:56 → C:\Users\matto
[DEBUG] 15:38:05.170 properties.go:GetString:28 → files
[TRACE] 15:38:05.170 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.170 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.170 shell.go:HasFiles(workspace.json) - 2.2065ms
[TRACE] 15:38:05.170 shell.go:Flags() - 0s
[TRACE] 15:38:05.170 shell.go:Pwd() - 0s
[TRACE] 15:38:05.170 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.170 shell_windows.go:56 → C:\Users\matto
[TRACE] 15:38:05.170 shell.go:GOOS() - 0s
[DEBUG] 15:38:05.170 segment.go:SetEnabled:512 → Segment: Os
[TRACE] 15:38:05.170 shell.go:GOOS() - 0s
[DEBUG] 15:38:05.170 properties.go:GetString:28 → 
[TRACE] 15:38:05.170 shell.go:Flags() - 0s
[TRACE] 15:38:05.170 shell.go:TemplateCache() - 0s
[TRACE] 15:38:05.170 shell.go:Flags() - 0s
[TRACE] 15:38:05.170 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.170 shell_windows.go:56 → C:\Users\matto
[TRACE] 15:38:05.170 shell.go:GOOS() - 0s
[TRACE] 15:38:05.170 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.170 segment.go:SetEnabled:512 → Segment: Battery
[DEBUG] 15:38:05.170 shell_windows.go:56 → C:\Users\matto
[TRACE] 15:38:05.170 shell.go:GOOS() - 0s
[TRACE] 15:38:05.170 shell_windows.go:IsWsl() - 0s
[DEBUG] 15:38:05.170 segment.go:SetEnabled:512 → Segment: Azfunc
[DEBUG] 15:38:05.170 properties.go:GetStringArray:52 → extensions: [host.json local.settings.json function.json]
[DEBUG] 15:38:05.170 properties.go:GetStringArray:52 → folders: []
[DEBUG] 15:38:05.170 properties.go:GetBool:22 → home_enabled: false
[TRACE] 15:38:05.170 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.170 shell_windows.go:56 → C:\Users\matto
[DEBUG] 15:38:05.170 properties.go:GetString:28 → files
[TRACE] 15:38:05.170 shell.go:Pwd() - 0s
[TRACE] 15:38:05.170 shell.go:Flags() - 0s
[TRACE] 15:38:05.170 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.170 shell_windows.go:56 → C:\Users\matto
[TRACE] 15:38:05.170 shell.go:GOOS() - 0s
[DEBUG] 15:38:05.170 shell.go:HasFiles:464 → false
[DEBUG] 15:38:05.170 segment.go:SetEnabled:512 → Segment: Aws
[TRACE] 15:38:05.170 shell.go:HasFiles(*.dart) - 2.2065ms
[TRACE] 15:38:05.170 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.170 properties.go:GetBool:22 → display_default: true
[DEBUG] 15:38:05.170 shell.go:Getenv:398 → NO DATA
[TRACE] 15:38:05.170 shell.go:Getenv(AWS_VAULT) - 0s
[DEBUG] 15:38:05.169 properties.go:GetBool:22 → home_enabled: false
[DEBUG] 15:38:05.170 shell.go:Getenv:398 → NO DATA
[TRACE] 15:38:05.170 shell.go:Getenv(AWS_DEFAULT_PROFILE) - 0s
[DEBUG] 15:38:05.170 shell.go:Getenv:398 → NO DATA
[TRACE] 15:38:05.170 shell.go:Pwd() - 0s
[TRACE] 15:38:05.170 shell.go:Getenv(AWS_PROFILE) - 0s
[DEBUG] 15:38:05.170 shell.go:Getenv:398 → NO DATA
[TRACE] 15:38:05.170 shell.go:Getenv(AWS_REGION) - 0s
[DEBUG] 15:38:05.170 shell_windows.go:56 → C:\Users\matto
[DEBUG] 15:38:05.170 shell.go:Getenv:398 → NO DATA
[TRACE] 15:38:05.170 shell.go:Getenv(AWS_DEFAULT_REGION) - 0s
[DEBUG] 15:38:05.170 properties.go:GetString:28 → files
[DEBUG] 15:38:05.170 shell.go:Getenv:398 → NO DATA
[TRACE] 15:38:05.170 shell.go:Pwd() - 0s
[TRACE] 15:38:05.170 shell.go:Getenv(AWS_CONFIG_FILE) - 0s
[DEBUG] 15:38:05.170 shell_windows.go:56 → C:\Users\matto
[ERROR] 15:38:05.170 shell.go:FileContent:533 → open C:\Users\matto/.aws/config: The system cannot find the path specified.
[TRACE] 15:38:05.170 shell.go:FileContent(C:\Users\matto/.aws/config) - 0s
[DEBUG] 15:38:05.171 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.171 shell.go:HasFiles(*.go) - 2.154ms
[TRACE] 15:38:05.171 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.171 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.171 shell.go:HasFiles(angular.json) - 2.8088ms
[DEBUG] 15:38:05.171 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.171 shell.go:HasFiles(package-lock.json) - 2.7215ms
[TRACE] 15:38:05.171 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.171 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.171 shell.go:HasFiles(Rakefile) - 1.6861ms
[TRACE] 15:38:05.171 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.171 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.171 shell.go:HasFiles(pom.xml) - 3.3763ms
[TRACE] 15:38:05.171 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.172 shell.go:HasFiles:464 → false
[DEBUG] 15:38:05.172 shell.go:HasFiles:464 → false
[DEBUG] 15:38:05.172 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.172 shell.go:HasFiles(*.cs) - 1.8283ms
[TRACE] 15:38:05.172 shell.go:HasFiles(host.json) - 1.8283ms
[TRACE] 15:38:05.172 shell.go:HasFiles(nx.json) - 1.8283ms
[TRACE] 15:38:05.172 shell.go:Pwd() - 0s
[TRACE] 15:38:05.172 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.172 shell.go:HasFiles:464 → false
[DEBUG] 15:38:05.172 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.172 shell.go:HasFiles(pubspec.yaml) - 1.8283ms
[TRACE] 15:38:05.172 shell.go:Pwd() - 0s
[TRACE] 15:38:05.172 shell.go:HasFiles(*.jl) - 1.8283ms
[DEBUG] 15:38:05.172 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.172 shell.go:HasFiles(*.rs) - 1.8283ms
[TRACE] 15:38:05.172 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.172 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.172 shell.go:HasFiles(go.mod) - 1.226ms
[DEBUG] 15:38:05.172 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.172 shell.go:HasFiles(Gemfile) - 1.1774ms
[DEBUG] 15:38:05.172 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.172 shell.go:HasFiles(package.json) - 1.1774ms
[TRACE] 15:38:05.172 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.172 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.172 shell.go:HasFiles(build.gradle.kts) - 1.1774ms
[TRACE] 15:38:05.172 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.173 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.173 shell.go:HasFiles(local.settings.json) - 1.0439ms
[TRACE] 15:38:05.173 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.173 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.173 shell.go:HasFiles(Cargo.toml) - 1.0439ms
[TRACE] 15:38:05.173 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.173 shell.go:HasFiles:464 → false
[DEBUG] 15:38:05.173 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.173 shell.go:HasFiles(*.csx) - 1.0439ms
[TRACE] 15:38:05.173 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.173 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.173 shell.go:HasFiles(*.js) - 525µs
[TRACE] 15:38:05.173 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.173 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.173 shell.go:HasFiles(build.sbt) - 525µs
[TRACE] 15:38:05.173 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.173 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.173 shell.go:HasFiles(Cargo.lock) - 515.7µs
[DEBUG] 15:38:05.173 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.173 shell.go:HasFiles(function.json) - 515.7µs
[DEBUG] 15:38:05.173 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.173 shell.go:HasFiles(pubspec.lock) - 515.7µs
[DEBUG] 15:38:05.173 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.173 shell.go:HasFiles(*.ts) - 515.7µs
[TRACE] 15:38:05.173 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.173 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.173 shell.go:HasFiles(.java-version) - 515.7µs
[TRACE] 15:38:05.173 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.173 shell.go:HasFolder:507 → false
[TRACE] 15:38:05.173 shell.go:HasFolder(.dart_tool) - 0s
[DEBUG] 15:38:05.173 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.173 shell.go:HasFiles(package.json) - 0s
[TRACE] 15:38:05.173 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.173 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.173 shell.go:HasFiles(.deps.edn) - 0s
[TRACE] 15:38:05.173 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.173 shell.go:CommandPath:614 → C:\Program Files\Python39\python.exe
[TRACE] 15:38:05.173 shell.go:CommandPath(python) - 5.5944ms
[DEBUG] 15:38:05.173 shell.go:HasFilesInDir:478 → false
[TRACE] 15:38:05.173 shell.go:HasFilesInDir(pyvenv.cfg) - 0s
[DEBUG] 15:38:05.173 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.173 shell.go:HasFiles(.nvmrc) - 0s
[TRACE] 15:38:05.173 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.173 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.173 shell.go:HasFiles(project.clj) - 0s
[TRACE] 15:38:05.173 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.173 shell.go:HasFilesInDir:478 → false
[TRACE] 15:38:05.173 shell.go:HasFilesInDir(pyvenv.cfg) - 0s
[DEBUG] 15:38:05.173 shell.go:Getenv:398 → NO DATA
[TRACE] 15:38:05.173 shell.go:Getenv(VIRTUAL_ENV) - 0s
[DEBUG] 15:38:05.173 shell.go:Getenv:398 → NO DATA
[TRACE] 15:38:05.173 shell.go:Getenv(CONDA_ENV_PATH) - 0s
[DEBUG] 15:38:05.173 shell.go:Getenv:398 → NO DATA
[TRACE] 15:38:05.173 shell.go:Getenv(CONDA_DEFAULT_ENV) - 0s
[DEBUG] 15:38:05.174 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.174 shell.go:HasFiles(pnpm-workspace.yaml) - 524.8µs
[DEBUG] 15:38:05.174 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.174 shell.go:Pwd() - 0s
[TRACE] 15:38:05.174 shell.go:HasFiles(build.boot) - 524.8µs
[TRACE] 15:38:05.174 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.174 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.174 shell.go:HasFiles(.pnpmfile.cjs) - 0s
[TRACE] 15:38:05.174 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.174 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.174 shell.go:HasFiles(*.java) - 0s
[TRACE] 15:38:05.174 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.174 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.174 shell.go:HasFiles(.npmrc) - 0s
[TRACE] 15:38:05.174 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.174 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.174 shell.go:HasFiles(*.class) - 0s
[TRACE] 15:38:05.174 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.174 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.174 shell.go:HasFiles(.vue) - 0s
[DEBUG] 15:38:05.174 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.174 shell.go:HasFiles(*.gradle) - 517.8µs
[TRACE] 15:38:05.174 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.174 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.174 shell.go:HasFiles(*.jar) - 0s
[TRACE] 15:38:05.174 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.174 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.174 shell.go:HasFiles(*.clj) - 0s
[TRACE] 15:38:05.174 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.174 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.174 shell.go:HasFiles(*.cljc) - 0s
[DEBUG] 15:38:05.176 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.176 shell.go:HasFiles(*.vb) - 3.623ms
[TRACE] 15:38:05.176 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.177 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.177 shell.go:HasFiles(*.sln) - 517.7µs
[TRACE] 15:38:05.177 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.177 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.177 shell.go:HasFiles(*.slnf) - 0s
[TRACE] 15:38:05.177 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.177 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.177 shell.go:HasFiles(*.csproj) - 0s
[TRACE] 15:38:05.177 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.177 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.177 shell.go:HasFiles(*.vbproj) - 520µs
[TRACE] 15:38:05.177 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.177 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.177 shell.go:HasFiles(*.fs) - 0s
[TRACE] 15:38:05.177 shell.go:Pwd() - 0s
[TRACE] 15:38:05.177 shell_windows_nix.go:BatteryState() - 7.5329ms
[DEBUG] 15:38:05.177 properties.go:GetString:28 → 
[TRACE] 15:38:05.177 shell.go:TemplateCache() - 0s
[DEBUG] 15:38:05.177 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.177 shell.go:HasFiles(*.fsx) - 0s
[TRACE] 15:38:05.177 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.177 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.177 shell.go:HasFiles(*.fsproj) - 0s
[TRACE] 15:38:05.177 shell.go:Pwd() - 0s
[DEBUG] 15:38:05.178 shell.go:HasFiles:464 → false
[TRACE] 15:38:05.178 shell.go:HasFiles(global.json) - 518.3µs
[DEBUG] 15:38:05.193 shell.go:CommandPath:614 → C:\Program Files\Docker\Docker\resources\bin\kubectl.exe
[TRACE] 15:38:05.193 shell.go:CommandPath(kubectl) - 23.547ms
[TRACE] 15:38:05.193 shell.go:HasCommand(kubectl) - 23.547ms
[ERROR] 15:38:05.257 shell.go:RunCommand:591 → exit status 1
[DEBUG] 15:38:05.257 shell.go:RunCommand:593 ↓
    error: current-context must exist in order to minify

[TRACE] 15:38:05.257 shell.go:RunCommand(kubectl config view --output yaml --minify) - 63.6248ms
[DEBUG] 15:38:05.257 properties.go:GetBool:22 → display_error: false
[DEBUG] 15:38:05.257 text.go:Render:70 → Rendering template:   {{ .CurrentDate | date .Format }}
[DEBUG] 15:38:05.257 text.go:Render:70 → Rendering template:  {{ if .WSL }}WSL at {{ end }}{{.Icon}}<#222222> </>
[DEBUG] 15:38:05.257 text.go:Render:70 → Rendering template:  {{ if not .Error }}{{ .Icon }}{{ .Percentage }}{{ end }}{{ .Error }} <#262626></>
[TRACE] 15:38:05.257 shell.go:TemplateCache() - 0s
[TRACE] 15:38:05.257 shell.go:TemplateCache() - 0s
[TRACE] 15:38:05.257 shell.go:TemplateCache() - 0s
[TRACE] 15:38:05.257 shell.go:Shell() - 0s
[TRACE] 15:38:05.257 shell.go:Shell() - 0s
[TRACE] 15:38:05.257 shell.go:Shell() - 0s
[DEBUG] 15:38:05.257 text.go:Render:70 → Rendering template: diamond
[DEBUG] 15:38:05.257 text.go:Render:70 → Rendering template: diamond
[DEBUG] 15:38:05.257 text.go:Render:70 → Rendering template: diamond
[DEBUG] 15:38:05.257 text.go:Render:70 → Rendering template: diamond
[DEBUG] 15:38:05.257 text.go:Render:70 → Rendering template: diamond
[DEBUG] 15:38:05.257 text.go:Render:70 → Rendering template: diamond
[DEBUG] 15:38:05.257 text.go:Render:70 → Rendering template: diamond
[DEBUG] 15:38:05.257 text.go:Render:70 → Rendering template: diamond
[DEBUG] 15:38:05.257 text.go:Render:70 → Rendering template: diamond
[DEBUG] 15:38:05.257 text.go:Render:70 → Rendering template: diamond
[DEBUG] 15:38:05.257 text.go:Render:70 → Rendering template: diamond
[DEBUG] 15:38:05.257 text.go:Render:70 → Rendering template: diamond
[DEBUG] 15:38:05.257 text.go:Render:70 → Rendering template: diamond
[DEBUG] 15:38:05.257 text.go:Render:70 → Rendering template: diamond
[DEBUG] 15:38:05.257 text.go:Render:70 → Rendering template: diamond
[DEBUG] 15:38:05.257 text.go:Render:70 → Rendering template: {{if eq "Charging" .State.String}}#b8e994{{end}}
[TRACE] 15:38:05.257 shell.go:TemplateCache() - 0s
[DEBUG] 15:38:05.257 text.go:Render:70 → Rendering template: {{if eq "Discharging" .State.String}}#fff34e{{end}}
[TRACE] 15:38:05.258 shell.go:TemplateCache() - 0s
[DEBUG] 15:38:05.258 text.go:Render:70 → Rendering template: {{if eq "Full" .State.String}}#33DD2D{{end}}
[TRACE] 15:38:05.259 shell.go:TemplateCache() - 0s
[DEBUG] 15:38:05.259 text.go:Render:70 → Rendering template: diamond
[DEBUG] 15:38:05.259 text.go:Render:70 → Rendering template: diamond
[TRACE] 15:38:05.259 shell.go:Shell() - 0s
[DEBUG] 15:38:05.259 shell_windows.go:TerminalWidth:109 → terminal width: 120
[TRACE] 15:38:05.259 shell_windows.go:TerminalWidth() - 586µs
[TRACE] 15:38:05.259 shell.go:Shell() - 0s
[TRACE] 15:38:05.259 shell.go:Shell() - 0s
[TRACE] 15:38:05.259 shell.go:Shell() - 0s
[TRACE] 15:38:05.259 shell.go:Flags() - 0s
[DEBUG] 15:38:05.259 properties.go:GetString:28 → {{ .Code }}
[TRACE] 15:38:05.259 shell.go:Pwd() - 0s
[TRACE] 15:38:05.259 shell.go:Flags() - 0s
[DEBUG] 15:38:05.259 shell_windows.go:56 → C:\Users\matto
[TRACE] 15:38:05.259 shell.go:Pwd() - 0s
[TRACE] 15:38:05.259 shell.go:GOOS() - 0s
[DEBUG] 15:38:05.259 segment.go:SetEnabled:512 → Segment: Status
[DEBUG] 15:38:05.259 shell_windows.go:56 → C:\Users\matto
[TRACE] 15:38:05.259 shell.go:StatusCodes() - 0s
[TRACE] 15:38:05.259 shell.go:GOOS() - 0s
[DEBUG] 15:38:05.259 text.go:Render:70 → Rendering template: {{ .Code }}
[DEBUG] 15:38:05.259 segment.go:SetEnabled:512 → Segment: Text
[TRACE] 15:38:05.259 shell.go:TemplateCache() - 0s
[TRACE] 15:38:05.259 shell.go:TemplateCache() - 0s
[DEBUG] 15:38:05.259 properties.go:GetBool:22 → always_enabled: true
[TRACE] 15:38:05.259 shell.go:TemplateCache() - 0s
[DEBUG] 15:38:05.259 text.go:Render:70 → Rendering template: 
[TRACE] 15:38:05.259 shell.go:Shell() - 0s
[DEBUG] 15:38:05.259 text.go:Render:70 → Rendering template: ╰─
[TRACE] 15:38:05.259 shell.go:Shell() - 0s
[DEBUG] 15:38:05.259 shell.go:Getenv:398 → NO DATA
[TRACE] 15:38:05.259 shell.go:Getenv(TERM_PROGRAM) - 0s
[DEBUG] 15:38:05.259 text.go:Render:70 → Rendering template: plain
[DEBUG] 15:38:05.259 text.go:Render:70 → Rendering template: {{ if gt .Code 0 }}#ef5350{{ end }}
[TRACE] 15:38:05.260 shell.go:TemplateCache() - 0s
[DEBUG] 15:38:05.260 text.go:Render:70 → Rendering template: plain
[TRACE] 15:38:05.260 shell.go:Shell() - 0s
[TRACE] 15:38:05.260 shell.go:Shell() - 0s
[TRACE] 15:38:05.260 shell.go:Shell() - 0s
[DEBUG] 15:38:05.260 shell.go:Getenv:398 → C:\Users\matto\AppData\Local
[TRACE] 15:38:05.260 shell.go:Getenv(LOCALAPPDATA) - 0s
[TRACE] 15:38:05.260 shell_windows.go:CachePath() - 0s
[TRACE] 15:38:05.260 shell.go:Flags() - 0s
```

--- a/src/segments/owm.go
+++ b/src/segments/owm.go
@@ -36,6 +36,8 @@ const (
 	CacheKeyResponse string = "owm_response"
 	// CacheKeyURL key used when caching the url responsible for the response
 	CacheKeyURL string = "owm_url"
+
+	PoshOWMAPIKey = "POSH_OWM_API_KEY"
 )
 
 type weather struct {
@@ -90,6 +92,14 @@ func (d *Owm) getResult() (*owmDataResponse, error) {
 	}
 
 	apikey := properties.OneOf[string](d.props, ".", APIKey, "apiKey")
+	if len(apikey) == 0 {
+		apikey = d.env.Getenv(PoshOWMAPIKey)
+	}
+
+	if len(apikey) == 0 {
+		return nil, errors.New("no api key found")
+	}
+
 	location := d.props.GetString(Location, "De Bilt,NL")
 	latitude := d.props.GetFloat64(Latitude, 91)    // This default value is intentionally invalid since there should not be a default for this and 0 is a valid value
 	longitude := d.props.GetFloat64(Longitude, 181) // This default value is intentionally invalid since there should not be a default for this and 0 is a valid value

--- a/src/segments/upgrade.go
+++ b/src/segments/upgrade.go
@@ -2,6 +2,7 @@ package segments
 
 import (
 	"encoding/json"
+	"errors"
 
 	"github.com/jandedobbeleer/oh-my-posh/src/build"
 	"github.com/jandedobbeleer/oh-my-posh/src/platform"
@@ -18,7 +19,10 @@ type Upgrade struct {
 	props properties.Properties
 	env   platform.Environment
 
+	// deprecated
 	Version string
+
+	UpgradeCache
 }
 
 const UPGRADECACHEKEY = "upgrade_segment"
@@ -33,43 +37,44 @@ func (u *Upgrade) Init(props properties.Properties, env platform.Environment) {
 }
 
 func (u *Upgrade) Enabled() bool {
-	current := build.Version
-	latest := u.cachedLatest(current)
-	if len(latest) == 0 {
-		latest = u.checkUpdate(current)
+	u.Current = build.Version
+	latest, err := u.cachedLatest(u.Current)
+	if err != nil {
+		latest, err = u.checkUpdate(u.Current)
 	}
 
-	if len(latest) == 0 || current == latest {
+	if err != nil || u.Current == latest.Latest {
 		return false
 	}
 
-	u.Version = latest
+	u.UpgradeCache = *latest
+	u.Version = u.Latest
 	return true
 }
 
-func (u *Upgrade) cachedLatest(current string) string {
+func (u *Upgrade) cachedLatest(current string) (*UpgradeCache, error) {
 	data, ok := u.env.Cache().Get(UPGRADECACHEKEY)
 	if !ok {
-		return "" // no cache
+		return nil, errors.New("no cache data")
 	}
 
 	var cacheJSON UpgradeCache
 	err := json.Unmarshal([]byte(data), &cacheJSON)
 	if err != nil {
-		return "" // invalid cache data
+		return nil, err // invalid cache data
 	}
 
 	if current != cacheJSON.Current {
-		return "" // version changed, run the check again
+		return nil, errors.New("version changed, run the check again")
 	}
 
-	return cacheJSON.Latest
+	return &cacheJSON, nil
 }
 
-func (u *Upgrade) checkUpdate(current string) string {
+func (u *Upgrade) checkUpdate(current string) (*UpgradeCache, error) {
 	tag, err := upgrade.Latest(u.env)
 	if err != nil {
-		return ""
+		return nil, err
 	}
 
 	latest := tag[1:]
@@ -79,7 +84,7 @@ func (u *Upgrade) checkUpdate(current string) string {
 	}
 	cacheJSON, err := json.Marshal(cacheData)
 	if err != nil {
-		return ""
+		return nil, err
 	}
 
 	oneWeek := 10080
@@ -87,5 +92,5 @@ func (u *Upgrade) checkUpdate(current string) string {
 	// update cache
 	u.env.Cache().Set(UPGRADECACHEKEY, string(cacheJSON), cacheTimeout)
 
-	return latest
+	return cacheData, nil
 }

--- a/website/docs/faq.mdx
+++ b/website/docs/faq.mdx
@@ -9,6 +9,21 @@ import TabItem from "@theme/TabItem";
 
 Before validating anything, make sure you're on the [latest version][latest] of Oh My Posh and your terminal and shell are up-to-date.
 
+### My prompt is broken after upgrading to PowerShell 7.4
+
+PowerShell 7.4 has an [issue][pwsh-20733] in encoding which affects the rendering of Oh My Posh.
+You can work around this by forcing the entire shell to UTF8.
+
+```powershell
+[Console]::OutputEncoding = [Text.Encoding]::UTF8
+oh-my-posh init pwsh --config ~/custom.omp.json | Invoke-Expression
+```
+
+:::warning
+Forcing the entire session to UTF8 can have unwanted side-effects for other executables. We can't be held liable for that,
+consider this a temporary workaround until the original PowerShell [issue][pwsh-20733] is resolved.
+:::
+
 ### I want to disable the upgrade notice
 
 You can disable the upgrade notice using the following command once:
@@ -28,7 +43,7 @@ oh-my-posh enable notice
 :::tip Windows
 The delay you're experiencing may be linked to the real-time protection features of antivirus software.
 
-To potentially alleviate this delay, ensure that Windows Defender or your primary antivirus software has an [exclusion][exclusion] set for the full path of the executable. To identify the full path of the executable in question, you can run the following command in a PowerShell prompt: 
+To potentially alleviate this delay, ensure that Windows Defender or your primary antivirus software has an [exclusion][exclusion] set for the full path of the executable. To identify the full path of the executable in question, you can run the following command in a PowerShell prompt:
 
 ```powershell
 (Get-Command oh-my-posh).Source
@@ -282,6 +297,7 @@ updated the icons themselves. A backup of the current config can be found in the
 
 This is a known problem with Xonsh. The issue is tracked [here][xonsh-issue].
 
+[pwsh-20733]: https://github.com/PowerShell/PowerShell/issues/20733
 [exclusion]: https://support.microsoft.com/en-us/windows/add-an-exclusion-to-windows-security-811816c0-4dfd-af4a-47e4-c301afe13b26
 [arch-terminfo]: https://wiki.archlinux.org/title/Bash/Prompt_customization#Terminfo_escape_sequences
 [ps-ansi-docs]: https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_ansi_terminals?view=powershell-7.2

--- a/website/docs/segments/owm.mdx
+++ b/website/docs/segments/owm.mdx
@@ -37,16 +37,16 @@ import Config from "@site/src/components/Config.js";
 
 ## Properties
 
-| Name            | Type      | Description                                                                                                                                                                                                        |
-| --------------- | --------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
-| `api_key`       | `string`  | Your API key from [Open Weather Map][owm]                                                                                                                                                                          |
-| `latitude`      | `float64` | The latitude of the requested location.                                                                                                                                                                            |
-| `longitude`     | `float64` | The longitude of the requested location.                                                                                                                                                                           |
+| Name            | Type      | Description                                                                                                                                                                                                          |
+| --------------- | --------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `api_key`       | `string`  | Your API key from [Open Weather Map][owm]. Can also be set using the `POSH_OWM_API_KEY` environment variable.                                                                                                        |
+| `latitude`      | `float64` | The latitude of the requested location.                                                                                                                                                                              |
+| `longitude`     | `float64` | The longitude of the requested location.                                                                                                                                                                             |
 | `location`      | `string`  | The requested location. Formatted as \<City,STATE,COUNTRY_CODE\>. City name, state code and country code divided by comma. Please, refer to ISO 3166 for the state codes or country codes - defaults to `DE BILT,NL` |
-| `units`         | `string`  | Units of measurement. Available values are standard (kelvin), metric (celsius), and imperial (fahrenheit) - defaults to `standard`                                                                                 |
-| `http_timeout`  | `int`     | The default timeout for http request is 20ms.                                                                                                                                                                      |
-| `cache_timeout` | `int`     | The default timeout for request caching is 10m. A value of 0 disables the cache.                                                                                                                                   |
-| `template`      | `string`  | A go [text/template][go-text-template] template extended with [sprig][sprig] utilizing the properties below - defaults to `{{.Weather}} ({{.Temperature}}{{.UnitIcon}})`                                           |
+| `units`         | `string`  | Units of measurement. Available values are standard (kelvin), metric (celsius), and imperial (fahrenheit) - defaults to `standard`                                                                                   |
+| `http_timeout`  | `int`     | The default timeout for http request is 20ms.                                                                                                                                                                        |
+| `cache_timeout` | `int`     | The default timeout for request caching is 10m. A value of 0 disables the cache.                                                                                                                                     |
+| `template`      | `string`  | A go [text/template][go-text-template] template extended with [sprig][sprig] utilizing the properties below - defaults to `{{.Weather}} ({{.Temperature}}{{.UnitIcon}})`                                             |
 
 ### Specifying Location
 

--- a/website/docs/segments/upgrade.mdx
+++ b/website/docs/segments/upgrade.mdx
@@ -39,8 +39,9 @@ import Config from "@site/src/components/Config.js";
 
 ### Properties
 
-| Name       | Type     | Description            |
-| ---------- | -------- | ---------------------- |
-| `.Version` | `string` | the new version number |
+| Name       | Type     | Description                         |
+| ---------- | -------- | ----------------------------------- |
+| `.Current` | `string` | the current version number          |
+| `.Latest`  | `string` | the latest available version number |
 
 [templates]: /docs/configuration/templates

